include/zephyr/bluetooth/hci.h

```diff
@@ -32,7 +32,11 @@ extern "C" {
  *
  * @return The string representation of the HCI error code.
  */
+#ifdef CONFIG_BT_DEBUG_HCI_ERR_TO_STR
 const char *bt_hci_err_to_str(uint8_t hci_err);
+#else
+static inline const char *bt_hci_err_to_str(uint8_t hci_err) { return ""; }
+#endif
```

Comment:
If the implementation of `bt_hci_err_to_str` just returned `""` instead of this "hack", does that increase memory usage significantly?

Reply:
It increases memory consumption a little bit. For the sample peripheral_hr on 52840dk with the default configuration (no debug logs):

With "optimization":
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      158376 B         1 MB     15.10%
             RAM:       25748 B       256 KB      9.82%
        IDT_LIST:          0 GB        32 KB      0.00%
```
Without (+16 bytes)
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      158392 B         1 MB     15.11%
             RAM:       25748 B       256 KB      9.82%
        IDT_LIST:          0 GB        32 KB      0.00%
```

Before this commit (-156)
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      158236 B         1 MB     15.09%
             RAM:       25748 B       256 KB      9.82%
        IDT_LIST:          0 GB        32 KB      0.00%
```

Reply:
Is that with `-Os`? 

I'm OK to spend 16 bytes of flash to just have a single definition of `bt_hci_err_to_str`

Reply:
Yes, this is with default compiler options